bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,323 @@
+            elif 'q=' not in search_link:
+                first_kata_div = choice(first_kata_div)
+            else:
+                first_kata_div = first_kata_div[0]
+
+            # there are numerous divs before arriving at the id of the kata, which can be used for the link.
+            first_kata_id = first_kata_div.a["href"].split("/")[-1]
+            return first_kata_id
+
+    async def kata_information(self, kata_id: str) -> Union[dict, Embed]:
+        """
+        Returns the information about the Kata.
+
+        Uses the codewars.com API to get information about the kata using the kata's ID.
+        """
+        async with self.bot.http_session.get(API_ROOT.format(kata_id=kata_id)) as response:
+            if response.status != 200:
+                error_embed = Embed(
+                    title=choice(NEGATIVE_REPLIES),
+                    description="We ran into an error with getting the kata information, try again later.",
+                    color=Colours.soft_red,
+                )
+                return error_embed
+
+            return await response.json()
+
+    @staticmethod
+    def main_embed(kata_information: dict) -> Embed:
+        """Creates the main embed which displays the description of the kata and the difficulty, along with the name."""
+        kata_description = kata_information["description"]
+        kata_url = f"https://codewars.com/kata/{kata_information['id']}"
+
+        # ensuring it isn't over the length 1024
+        if len(kata_description) > 1024:
+            kata_description = "\n".join(kata_description[:1000].split("\n")[:-1])
+            kata_description += f"\n[Read more...]({kata_url})"
+
+        kata_embed = Embed(
+            title=kata_information['name'],
+            description=kata_description,
+            color=Color.from_rgb(
+                *MAPPING_OF_KYU[int(kata_information['rank']['name'].replace(' kyu', ''))]
+            ),
+            url=kata_url,
+        )
+        kata_embed.add_field(name="Difficulty", value=kata_information['rank']['name'], inline=False)
+        return kata_embed
+
+    @staticmethod
+    def language_embed(kata_information: dict) -> Embed:
+        """Creates the 'language embed' which displays all languages the kata supports."""
+        languages = '\n'.join(map(str.title, kata_information['languages']))
+        language_embed = Embed(
+            title="Languages Supported",
+            description=f"```nim\n{languages}```",
+            color=Colours.python_blue,
+        )
+        return language_embed
+
+    @staticmethod
+    def tags_embed(kata_information: dict) -> Embed:
+        """
+        Creates the 'tags embed' which displays all the tags of the Kata.
+
+        Tags explain what the Kata is about, this is what codewars.com calls categories.
+        """
+        tags = '\n'.join(kata_information['tags'])
+        tags_embed = Embed(
+            title="Tags",
+            description=f"```nim\n{tags}```",
```

Comment:
Does this not need to be followed by a newline as well?

Reply:
Can you elaborate?

Reply:
Well, after `{tags}` do we not need a new line so that it becomes (demonstrated with single-quotes)
```
'''nim
{tags}
'''
```
Instead of what this now becomes
```
'''nim
{tags}'''
```
But that may not matter?